lib/api/CInferenceModelMetadata.cc

```diff
@@ -43,7 +43,12 @@ void CInferenceModelMetadata::writeTotalFeatureImportance(TRapidJsonWriter& writ
             for (std::size_t j = 0; j < m_ClassValues.size(); ++j) {
                 writer.StartObject();
                 writer.Key(JSON_CLASS_NAME_TAG);
-                writer.String(m_ClassValues[j]);
+                if (m_PredictionFieldTypeResolverWriter) {
```

Comment:
It always feels cleaner to me to default these sorts of call backs to avoid having if not null in the code. It also protects against accidental misuse. You can declare this very easily with a lambda inline in the class definition or you could create a `defaultPredictionFieldTypeResolverWriter` private member function to default initialise in the same way. For example:
```
TPredictionFieldTypeResolverWriter m_PredictionFieldTypeResolverWriter = [](const std::string& value, TRapidJsonWriter& writer) { writer.String(m_ClassValues[j]); };
```